src/gui/chunk_panel.rs

```diff
@@ -1,43 +1,67 @@
 // use std::mem::replace;
 
+use std::collections::HashMap;
+use std::mem::replace;
+
+use crate::event::{Event, EventGenerator};
 use crate::forza::{self, Lap};
+use crate::forza::{ChunkId, ChunkSelector, LapId};
 use eframe::egui;
 
-pub type ChunkID = usize;
-pub type LapID = Option<u16>;
-
-#[derive(PartialEq, Default, Clone, Copy)]
-pub struct ChunkSelection(pub ChunkID, pub LapID);
+pub enum ChunkPanelEvent {
+    ChangeSelection(ChunkSelector),
+    RemoveChunk(ChunkSelector),
+}
 
-#[derive(Default)]
 pub struct ChunkPanel {
-    pub selection: ChunkSelection,
-    pub trash_chunk: Option<ChunkSelection>,
+    selection: ChunkSelector,
+    events: HashMap<u8, Event>,
 }
 
 impl ChunkPanel {
-    fn select(&mut self, chunk_id: ChunkID, lap_id: LapID) {
-        self.selection = ChunkSelection(chunk_id, lap_id)
+    pub fn new() -> Self {
+        Self {
+            selection: ChunkSelector::default(),
+            events: HashMap::with_capacity(1),
+        }
+    }
+
+    fn select(&mut self, chunk_id: ChunkId, lap_id: LapId) {
+        self.selection = ChunkSelector(chunk_id, lap_id);
```

Comment:
Is this still needed if the selection is passed within the event?

Reply:
Correct, it still not required

Reply:
It's already used to generate the Event and update self.selection.